apps/vaporgui/BarbEventRouter.cpp

```diff
@@ -1,131 +1,45 @@
-	qsapp->setWidgetResizable(true);
-	addTab(qsapp,"Appearance");
-
-	_geometry = new BarbGeometrySubtab(this);
-	QScrollArea *qsgeo = new QScrollArea(this);
-	qsgeo->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
-	qsgeo->setWidget(_geometry);
-	qsgeo->setWidgetResizable(true);
-	addTab(qsgeo, "Geometry");
-	
-	_annotation = new BarbAnnotationSubtab(this);
-	QScrollArea *qsAnnotation = new QScrollArea(this);
-	qsAnnotation->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
-	qsAnnotation->setWidget(_annotation);
-	qsAnnotation->setWidgetResizable(true);
-	addTab(qsAnnotation, "Annotation");
-}
-
-void BarbEventRouter::GetWebHelp(
-    vector <pair <string, string> > &help
-) const {
-    help.clear();
-
-    help.push_back(make_pair(
-		"Barb Overview",
-		"http://www.vapor.ucar.edu/docs/vapor-gui-help/BarbOverview"
-    )); 
-    help.push_back(make_pair(
-		"Renderer control",
-		"http://www.vapor.ucar.edu/docs/vapor-how-guide/renderer-instances"
-    )); 
-    help.push_back(make_pair(
-		"Data accuracy control",
-		"http://www.vapor.ucar.edu/docs/vapor-how-guide/refinement-and-lod-control"
-    )); 
-    help.push_back(make_pair(
-		"Barb geometry options",
-		"http://www.vapor.ucar.edu/docs/vapor-gui-help/BarbGeometry"
-    )); 
-    help.push_back(make_pair(
-		"Barb Appearance settings","<>"
-		"http://www.vapor.ucar.edu/docs/vapor-gui-help/BarbAppearance"
-    )); 
-}
-
-void BarbEventRouter::_initializeTab() {
-	_updateTab();
-	BarbParams* rParams = (BarbParams*) GetActiveParams();
-	DataMgr* dataMgr = GetActiveDataMgr();
-
-	_variables->Initialize(rParams, dataMgr);
-}
-
-void BarbEventRouter::_updateTab(){
-	//The variable tab updates itself:
-	_variables->Update(
-		GetActiveDataMgr(),
-		_controlExec->GetParamsMgr(),
-		GetActiveParams()
-	);
-
-	_appearance->Update(
-		GetActiveDataMgr(),
-		_controlExec->GetParamsMgr(),
-		GetActiveParams()
-	);
-	
-	_geometry->Update(
-		_controlExec->GetParamsMgr(),
-		GetActiveDataMgr(),
-		GetActiveParams()
-	);
-
-	_annotation->Update(
-		_controlExec->GetParamsMgr(),
-		GetActiveDataMgr(),
-		GetActiveParams()
-	);
+    AddSubtab("Variables", new PGroup({
+        new PSection("Variable Selection", {
+            new S::PDimensionSelector,
+            new S::PXFieldVariableSelector,
+            new S::PYFieldVariableSelector,
+            new S::PZFieldVariableSelector,
+            new S::PHeightVariableSelector,
+            new S::PColorMapVariableSelector,
+        }),
+        new PFidelitySection
+    }));
+    
+    AddSubtab("Appearance", new PGroup({
+        new PSection("Barbs", {
+            (new PIntegerSliderEdit(BP::_xBarbsCountTag, "X Barbs"))->SetRange(1, 50),
+            (new PIntegerSliderEdit(BP::_yBarbsCountTag, "Y Barbs"))->SetRange(1, 50),
+            (new PIntegerSliderEdit(BP::_zBarbsCountTag, "Z Barbs"))->SetRange(1, 50),
+            (new PDoubleSliderEdit(BP::_lengthScaleTag, "Length Scale"))->SetRange(0.01, 4)->EnableDynamicUpdate(),
```

Comment:
Does EnableDynamicUpdate enable the user to change the range of the slider? I don't seem to be able to change it. How is it done?

Reply:
`EnableDynamicUpdate()` causes "fast" renders to occur as the user moves the slider for live updates. The slider-edit `VWidget`  itself does not support changing the range.

Reply:
By "fast" renders do you  mean the callback isn't called until the slider is released?

Do we have support for changing the range? I thought we did and this is certainly a place where it would be useful.

Reply:
The intermediate updates change the params without adding to the undo history and Render(fast=true) is called. 

The two control range slider I created supports this but VSliderEdit does not support changing the range.

Reply:
@stasj Can you add support to VSliderEdit? There are a number of instances where this would be useful.

Reply:
@clyne @stasj VIntSliderEdit and VDoubleSliderEdit support range modification, FYI.

Reply:
@sgpearse I figured that out eventually. Why do we still have VSliderEdit? It is not documented there that we should be using the other classes.

@clyne done.

Reply:
@stasj has enabled user changeable ranges, and will add an option to optionally disable it.

Reply:
Done.